dvc/checkout.py

```diff
@@ -217,7 +217,7 @@ def _checkout_dir(
         or modified
     )
 
-    fs.state.save(path_info, obj.hash_info)
+    fs.repo.state.save(path_info, fs, obj.hash_info)
```

Comment:
Looks a bit odd with `fs.repo`, but we'll be getting rid of `fs.repo` in the folloups and it will be cleaner. For now detaching the sate from fs.